### PR TITLE
init graph debug

### PR DESCRIPTION
**Pull request by Superjomn**
- draw a graph for Block (due to loacl scope, only block(0) is supported)
- highlights some symbols using regex. 

```python
import paddle.v2.fluid.debuger as debuger

debuger.draw_block_graphviz(framework.default_main_program().block(0),
                            highlights=["learning_rate*"])
```

image

### CODE REVIEW

File: python/paddle/v2/fluid/framework.py [hunk 1]
```diff
@@ -1123,3 +1122,59 @@ def program_guard(main_program, startup_program=None):
     switch_main_program(main_program)
     if startup_program is not None:
         switch_startup_program(startup_program)
+
+
+def debug_block_image(block, highlights=[], path="./temp.dot"):
```
Comment (jacquesqiao) [inline python/paddle/v2/fluid/framework.py]: I think it's better to move this function to an independent file

Comment (Superjomn) [reply python/paddle/v2/fluid/framework.py]: fluid.debuger ?

Comment (jacquesqiao) [reply python/paddle/v2/fluid/framework.py]: cool, I think that is ok.

File: python/paddle/v2/fluid/debuger.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from graphviz import GraphPreviewGenerator
+import proto.framework_pb2 as framework_pb2
+
+
+def draw_block_graphviz(block, highlights=[], path="./temp.dot"):
```
Comment (jacquesqiao) [inline python/paddle/v2/fluid/debuger.py]: do not use collections as default value

**Comment by jacquesqiao** [comment]
I think we can have a directory called debugger, and put all the related file into this directory~

**Pull request merged**